Python/bytecodes.c

```diff
@@ -673,7 +673,7 @@ dummy_func(
                 err = 1;
             }
             else {
-                err = PyObject_SetItem(PyStackRef_AsPyObjectBorrow(container), slice, PyStackRef_AsPyObjectSteal(v));
+                err = PyObject_SetItem(PyStackRef_AsPyObjectBorrow(container), slice, PyStackRef_AsPyObjectBorrow(v));
```

Comment:
`PyObject_SetItem` and the entire family actually takes borrowed references to the container and item. So this steal is wrong, it should be borrow.

This was caught when the `PyStackRef_CLOSE(v)` below tried to close an already invalid stackref (it was made invalid by the steal).